#!/usr/bin/env python
PACKAGE = "pd_controller"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()


gen.add("c_prop_alt",              double_t, 0,    "const for P-part of altitude",                   3,     0, 5)
gen.add("c_int_alt",              double_t, 0,    "const for I-part of  altitude",                   0.1,  0   , 2)
gen.add("c_deriv_alt",              double_t, 0,    "const for D-part of  altitude",                   1,     0, 2)
gen.add("c_prop_trans",              double_t, 0,    "const for P-part of translation",                   1.5,     0, 2)
gen.add("c_int_trans",              double_t, 0,    "const for I-part of translation",                   0.10,  0   , 2)
gen.add("c_deriv_trans",              double_t, 0,    "const for D-part of translation",                   0.4,     0, 2)

gen.add("c_prop_yaw",              double_t, 0,    "const for P-part of translation",                   0.15,     0, 2)
gen.add("c_int_yaw",              double_t, 0,    "const for I-part of translation",                   0,     0, 2)
gen.add("c_deriv_yaw",              double_t, 0,    "const for D-part of translation",                   0,     0, 2)


exit(gen.generate(PACKAGE, "pd_controller", "my_param"))
